src/gc/config.d

```diff
@@ -62,70 +66,169 @@ struct Config
-                else if(s == "concurrent")
-                    concurrent = v != 0;
-                else if(s == "initReserve")
-                    initReserve = v;
-                else if(s == "minPoolSize")
-                    minPoolSize = v;
-                else if(s == "maxPoolSize")
-                    maxPoolSize = v;
-                else if(s == "incPoolSize")
-                    incPoolSize = v;
-                else
+                static if (!is(typeof(__traits(getMember, this, field)) == function))
                 {
-                    printf("Unknown GC option \"%.*s\"\n", cast(int) s.length, s.ptr);
-                    return false;
+                case field:
+                    if (!parse(name, tail, __traits(getMember, this, field)))
+                        return false;
+                    break;
                 }
-                p = r;
             }
-            else
-            {
-                printf("Incomplete GC option \"%.*s\"\n", cast(int) s.length, s.ptr);
-                return false;
+            break;
+
+            default:
+                return optError("Unknown", name);
             }
+            opt = skip!isspace(tail);
         }
         return true;
     }
 }
+
+private:
+
+bool optError(in char[] msg, in char[] name)
+{
+    version (unittest) if (inUnittest) return false;
+
+    fprintf(stderr, "%.*s GC option '%.*s'.\n",
+            cast(int)msg.length, msg.ptr,
+            cast(int)name.length, name.ptr);
+    return false;
+}
+
+inout(char)[] skip(alias pred)(inout(char)[] str)
+{
+    return find!(c => !pred(c))(str);
+}
+
+inout(char)[] find(alias pred)(inout(char)[] str)
+{
+    foreach (i; 0 .. str.length)
+        if (pred(str[i])) return str[i .. $];
+    return null;
+}
+
+bool parse(T:size_t)(const(char)[] optname, ref const(char)[] str, ref T res)
+in { assert(str.length); }
+body
+{
+    size_t i, v;
+    for (; i < str.length && isdigit(str[i]); ++i)
+        v = 10 * v + str[i] - '0';
+
+    if (!i)
+        return parseError("a number", optname, str);
+    str = str[i .. $];
+    res = v;
+    return true;
+}
+
+bool parse(T:bool)(const(char)[] optname, ref const(char)[] str, ref T res)
+in { assert(str.length); }
+body
+{
+    if (str[0] == '1' || str[0] == 'y' || str[0] == 'Y')
+        res = true;
+    else if (str[0] == '0' || str[0] == 'n' || str[0] == 'N')
+        res = false;
+    else
+        return parseError("'0/n/N' or '1/y/Y'", optname, str);
+    str = str[1 .. $];
+    return true;
+}
+
+bool parse(T:float)(const(char)[] optname, ref const(char)[] str, ref T res)
+in { assert(str.length); }
+body
+{
+    char[8] buf=void;
+
+    auto tail = skip!isdigit(str);
+    if (tail.length && tail[0] == '.')
+        tail = skip!isdigit(tail[1 .. $]);
+    auto opt = str[0 .. $ - tail.length];
+    immutable len = min(buf.length - 1, opt.length);
+    buf[0 .. len] = opt[0 .. len];
+    buf[len] = '\0';
+
+    if (sscanf(buf.ptr, "%f", &res) != 1)
```

Comment:
Maybe play with sscanf's width options so you don't need a zero-terminated string?